### DISCUSSION TITLE
Signing and Notarizing: Windows and macOS

### DISCUSSION THREAD

**Discussion by rfay** [goreleaser/goreleaser#3153] [Q&A] [answered]
I can build all binaries on all platforms but...
* With an EV cert for Windows, I can only sign on Windows
* For macOS binaries signing and notarization is really most predictably done on macOS.

I use GitHub Actions for building and releasing, and currently have a complex flow where most binaries are built on Linux (Github Actions Hosted Runner), then the macOS binaries are signed on a macOS Github Actions Hosted Runner, and in the meantime Windows binaries are built and signed on a Windows self-hosted runner.

<img width="990" alt="_vendor__Routine_go_dependency_update_to_satisfy_dependabot___3899__·_drud_ddev_0b49560" src="https://user-images.githubusercontent.com/112444/173165728-c6bba066-651a-4ad6-97c3-ae3e1c96fd7a.png">

I'd be interested in suggestions on how I can build all these things in a natural way with goreleaser (OS or Pro, either way). 

One way to do it is to keep the current flow and then use skip-build in goreleaser for the uploading parts and things like that.

I'd sure appreciate other thoughts on how this can be done.

**Comment by caarlos0** [comment, marked as answer]
I'm afraid there's no way to do this without getting a bit complex... one possible solution:

- one goreleaser files just for the builds, so you just `goreleaser build` in Linux
- sign the binaries in windows and macos
- another goreleaser config that uses the prebuilt builder to import the binaries, signed or not, and then you do the final release steps (creating packages, archives, etc)

Here's an example of spliting the binary build and release - no signing though: https://github.com/caarlos0/goreleaser-pro-prebuilt-example

Hope this helps!

**Reply by rfay** [reply]
- Thanks - that's better than just keeping my build stage as is and using goreleaser for the release. Unfortunately everything does get scattered all over the place. Back before Apple required notarization (and before DDEV had an EV cert) all this could be done on linux. Oh those easy days :)

**Reply by caarlos0** [reply]
yes, those things seem to keep getting worse as everyone seems to invent their own thing that only works on their OS 😞 

I'm not sure if there's much we could do on goreleaser to improve all that... maybe a "full" example with github actions and all those things I just mentioned in the previous comment... although I also don't have the certs for those things... I should probably get the developer account and try to do that some day.

**Reply by rfay** [reply]
It's off-topic, but I do think it's worth making sure that all our artifacts are signed and notarized. It builds confidence from the user and they never have to learn weird workarounds to open unsigned executables.

**Reply by caarlos0** [reply]
yes, its the ideal world :)
